Engineering/Mobile/Flow in Kotlin.md

```diff
@@ -0,0 +1,263 @@
+- `CollectLatest` in the order hand, only alert you the last data the source emitted.
+
+**Diff between Collect and CollectLatest**
+
+You may wonder what is the different between `Collect` and `CollectLatest`. 
+- The behavior of `Collect` is that while the collector is doing his work, all emitting work will be pending. 
+![PendingWhileCollect.png](https://khanhth-public-image-raw.s3.ap-southeast-1.amazonaws.com/work/brainery-assets/CountDownCollectPendin+(1).png)
+- But `CollectLatest` forcing the collector to start over.
+![CollectLatest](https://khanhth-public-image-raw.s3.ap-southeast-1.amazonaws.com/work/brainery-assets/CoutnDownCollectLatest+(1).png)
+
+So using `CollectLatest` on UI observation will be the better choice to reflect the state.
+        
+### Flow operations
+
+In real life, we can see a lot of different cases/problems, and not many of them can resolve in the same way. So let me introduce you to some of the operators that can help to organize data.
+
+- **Intermediate operations** are the operations that won't invoke the stream, instead its functionality will modify the stream. We have some operations like:
+    - `Filter` operator, allow you to filter out what you don’t need downstream by returning bool value to the filter lambda.
+    - `Map` in flow is different from the map in RX. When map in RX can only transform the `data type`, map in flow can both transform `data value`, and `data type`
+    - `onEach` operator is going to notify you whenever an item is emitted from a flow, but it won’t start the cold flow like collect should be
+    - `Catch` collect operator won’t receive any call whenever an error happens. Instead catch will be notified with an exception.
+    - `Buffer` is working like a `queue` when `flow` emits quicker than the `collector` can consume, we will buffer it under the queue and execute it in `FIFO` order. The `configuration` allows us to control `capacity` and if the `boundary limit` reaches, we can drop or suspend the queue.
+    - `Conflate` similar to buffer but the collector will only get the latest data available, and skipped the rest if the collector cannot finish his work at the time.
+    - `Collect Latest` is the harshest. It works similarly with `conflate`, except it won't wait for a `collector` to finish his job. Instead, it force the collector to start the job from the beginning.
+- **Terminal operations**, this kind of operation will invoke the flow.
+    - `Count` return a number of item that match the term
+    - `Reduce` whenever a new data emitted, we can access to the `Accumulator` variable of previous calculation.
+    - `Fold` is work the same as `Reduce` but support initial value
+    - `FlatMap`
+        - `flatMapConcat`, a basic flatten function. Use this when you want to flattening a `flow` in a `flow`
+        - `flatMapMerge` execute all the flows at the same time.
+        - `flatMapLatest` is like `collectLatest`, it only wait for the last item emitted
+
```

Comment:
Not sure if is it all operators of Flow but it is not so much compared to other Reactive frameworks. If you planning to write other articles to cover them all I have no complaints. But if this is the only one I suggested adding other.